src/char/basic_ranged.py

```diff
@@ -0,0 +1,216 @@
+import keyboard
+from utils.custom_mouse import mouse
+from char import IChar
+from template_finder import TemplateFinder
+from ui import UiManager
+from pather import Pather
+from logger import Logger
+from screen import Screen
+from utils.misc import wait, cut_roi
+import time
+import random
+import numpy as np
+from pather import Pather, Location
+
+
+class Basic_Ranged(IChar):
+    def __init__(self, skill_hotkeys, char_config, screen: Screen, template_finder: TemplateFinder, ui_manager: UiManager, pather: Pather):
+        Logger.info("Setting up Basic Ranged Character")
+        super().__init__(skill_hotkeys, char_config, screen, template_finder, ui_manager)
+        self._pather = pather
+        self._do_pre_move = True
+        if not self._skill_hotkeys["teleport"]:
+            self._do_pre_move = False
```

Comment:
I dont think we need this here. This is just specifically for walking pala to switch from vigor to concentration for the last node traverse as we already attack quite a bit. Such a specific thing shouldn't be done here imho.